### PR TITLE
152 add parallel coordinates plot

### PR DESCRIPTION
**Pull request by nmaarnio**
Implemented a parallel coordinates plot using matplotlib and seaborn. The tool takes in a dataframe and draws connected lines horizontally through the numerical feature space of the dataframe. A column used for color-encoding can be specified if wanted, and the column can be numerical or categorical.

NaN-values need to be removed before plotting currently, this could be changed if wanted.

Styling of the plot is currently limited to setting a title and selecting a color palette.

### CODE REVIEW

File: eis_toolkit/exploratory_analyses/parallel_coordinates.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+    _plot_parallel_coordinates(
+        data=data,
+        data_labels=data_labels,
+        color_data=color_data,
+        color_column_name=color_column_name,
+        plot_title=plot_title,
+        palette_name=palette_name,
+        curved_lines=curved_lines,
+    )
```
Comment (msorvoja) [inline eis_toolkit/exploratory_analyses/parallel_coordinates.py:192]: The outer function should also return the figure.

File: eis_toolkit/exploratory_analyses/parallel_coordinates.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        axis.spines["top"].set_visible(False)
+        axis.spines["bottom"].set_visible(False)
+        if axis != main_axis:
+            axis.spines["right"].set_visible(False)
+            axis.yaxis.set_ticks_position("left")
+            axis.spines["left"].set_position(("axes", i / (normalized_data.shape[1] - 1)))
+
+    main_axis.set_xlim(0, normalized_data.shape[1] - 1)
+    main_axis.set_xticks(range(normalized_data.shape[1]))
+    main_axis.set_xticklabels(data_labels, fontsize=14)
```
Comment (msorvoja) [inline eis_toolkit/exploratory_analyses/parallel_coordinates.py:82]: Longer labels overlap (see penguins plot in the notebook). How about rotating and decreasing fontsize?
```suggestion
    main_axis.set_xticklabels(data_labels, fontsize=10, rotation=45)
```
Screenshot from 2023-07-25 07-58-49

Comment (msorvoja) [reply eis_toolkit/exploratory_analyses/parallel_coordinates.py:82]: Or increase figure size (`fig, main_axis = plt.subplots(figsize=(15, 15)` at row 52, for example)

**Review by msorvoja — COMMENTED** [review]
Looks good overall! Just a few comments and suggestions.

File: eis_toolkit/exploratory_analyses/parallel_coordinates.py (hunk 2 above)
Comment (nmaarnio) [reply eis_toolkit/exploratory_analyses/parallel_coordinates.py:82]: I only changed to fontsize to 10 for now, that seemed to give already quite a lot more space for the feature labels.

**Review by msorvoja — APPROVED** [review]
Looks good now!

**Pull request merged**